### PR TITLE
[HOPSWORKS-2763] Opensearch 1.3.1

### PR DESCRIPTION
**Pull request by jimdowling**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-2763

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -13,8 +12,9 @@
 #
 # Logstash
 #
-default['logstash']['version']                               = "7.2.0"
-default['logstash']['url']                                   = "#{node['download_url']}/logstash-oss-#{node['logstash']['version']}.tar.gz"
+default['logstash']['version']                               = "7.13.2"
+default['logstash']['url']                                   = "#{node['download_url']}/opensearch/logstash-oss-with-opensearch-output-plugin-#{node['logstash']['version']}-linux-x64.tar.gz"
+#default['logstash']['sha512']                                = "de5c7ee0d1296787032d91733bb18d6cb9669e8887e683930f9d9c285b28e582b4b4aaf2e8e2365283496e71d00baec8dd109f532170f0e0cc88d35497f79424"
```
Comment (maismail) [inline attributes/default.rb:17]: remove sha512 if not used

File: attributes/default.rb [hunk 2]
```diff
@@ -58,8 +58,9 @@
 #
 # Filebeat
 
-default['filebeat']['version']                  = "7.2.0"
-default['filebeat']['url']                      = "#{node['download_url']}/filebeat-oss-#{node['filebeat']['version']}-linux-x86_64.tar.gz"
+default['filebeat']['version']                  = "7.12.1"
```
Comment (maismail) [inline attributes/default.rb:61]: why not `7.13.2` same as logstash?
https://www.elastic.co/downloads/past-releases/filebeat-oss-7-13-2

File: attributes/default.rb [hunk 3]
```diff
@@ -81,13 +82,13 @@
...
 default['hopslog']['public_ips']          = ['10.0.2.15']
 
-# Kibana Opendistro Security plugin
-default['kibana']['opendistro_security']['url']                                   = "#{node['download_url']}/opendistro_security_kibana_plugin-#{node['elastic']['opendistro']['version']}.zip"
-default['kibana']['opendistro_security']['https']['enabled']                      = "true"
-default['kibana']['opendistro_security']['multitenancy']['global']['enabled']     = "false"
-default['kibana']['opendistro_security']['multitenancy']['private']['enabled']    = "true"
-default['kibana']['opendistro_security']['cookie']['ttl']                         = node['elastic']['opendistro_security']['jwt']['exp_ms'].to_i
+# Kibana Opensearch Security plugin
+default['kibana']['opensearch_security']['url']                                   = "#{node['download_url']}/opensearch_security_kibana_plugin-#{node['elastic']['opensearch']['version']}.zip"
```
Comment (maismail) [inline attributes/default.rb:86]: why do you need the plugin isn't it installed by default as part of opensearch_dashboard?

File: attributes/default.rb [hunk 4]
```diff
@@ -81,13 +82,13 @@
...
+default['kibana']['opensearch_security']['https']['enabled']                      = "true"
+default['kibana']['opensearch_security']['multitenancy']['global']['enabled']     = "false"
+default['kibana']['opensearch_security']['multitenancy']['private']['enabled']    = "true"
+default['kibana']['opensearch_security']['cookie']['ttl']                         = node['elastic']['opensearch_security']['jwt']['exp_ms'].to_i
 # the session ttl time is set to be twice the time the cookie ttl time, in order to solve the session expiry issue in kibana_addr
-# https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/issues/31
-default['kibana']['opendistro_security']['session']['ttl']                        = 2 * node['elastic']['opendistro_security']['jwt']['exp_ms'].to_i
-default['kibana']['opendistro_security']['session']['keepalive']                  = "true"
+# https://github.com/opensearch-for-elasticsearch/security-kibana-plugin/issues/31
```
Comment (maismail) [inline attributes/default.rb:92]: Link doesn't work anymore, i wonder if this is an issue anymore though, will need to test it
https://github.com/opensearch-project/security-dashboards-plugin/pull/560
```suggestion
# https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/issues/31
```

File: recipes/install.rb [hunk 5]
```diff
@@ -206,6 +206,7 @@
         user "root"
         code <<-EOH
                 tar -xf #{cached_package_filename} -C #{node['hopslog']['dir']}
+                cd #{node['hopslog']['dir']}
```
Comment (maismail) [inline recipes/install.rb:209]: ?

File: templates/default/opensearch_dashboards.yml.erb [hunk 6]
```diff
@@ -29,18 +29,18 @@ server.rewriteBasePath: false
 
 # The URLs of the Elasticsearch instances to use for all your queries.
```
Comment (maismail) [inline templates/default/opensearch_dashboards.yml.erb:30]: you should base on this file 
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.1/config/opensearch_dashboards.yml

File: templates/default/opensearch_dashboards.yml.erb [hunk 7]
```diff
@@ -29,18 +29,18 @@ server.rewriteBasePath: false
...
 # that connects to this Kibana instance.
 #elasticsearch.preserveHost: true
-elasticsearch.preserveHost: true
+opensearch.preserveHost: true
 
 # Kibana uses an index in Elasticsearch to store saved searches, visualizations and
 # dashboards. Kibana creates a new index if the index doesn't already exist.
 #kibana.index: ".kibana"
-kibana.index: "<%= node['elastic']['opendistro_security']['kibana']['index'] %>"
+kibana.index: "<%= node['elastic']['opensearch_security']['kibana']['index'] %>"
```
Comment (maismail) [inline templates/default/opensearch_dashboards.yml.erb:43]:
```suggestion
opensearchDashboards.index: "<%= node['elastic']['opensearch_security']['kibana']['index'] %>"
```

File: templates/default/start-kibana.sh.erb [hunk 8]
```diff
@@ -10,6 +10,7 @@ if [ -f $PID_FILE ] ; then
  fi
 fi
 
+export JAVA_HOME=<%= node['elastic']['base_dir'] %>/jdk
```
Comment (maismail) [inline templates/default/start-kibana.sh.erb:13]: no need it is a node application

File: templates/default/start-kibana.sh.erb [hunk 9]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-PID_FILE=<%= node['kibana']['pid_file'] %>
+export PID_FILE=<%= node['kibana']['pid_file'] %>
```
Comment (maismail) [inline templates/default/start-kibana.sh.erb:3]: why aren't you changing the service name to opensearch_dashboard?

File: attributes/default.rb [hunk 10]
```diff
@@ -39,12 +39,12 @@
...
-default['kibana']['url']                        = "#{node['download_url']}/kibana-oss-#{node['kibana']['version']}-linux-x86_64.tar.gz"
+default['kibana']['version']                    = "1.1.0"
+default['kibana']['url']                        = "#{node['download_url']}/opensearch/opensearch-dashboards-#{node['kibana']['version']}-linux-x64.tar.gz"
+#default['kibana']['sha512']                     = "57c3b59b8f5970e781f2ea78db98af9d0b0ff183dcb15f2e87b8ff29098704ab08b093c0bd2dcb05ea01883a0c49572d54800d8789e5398c36b2aa1f56179ba2"
 default['kibana']['port']                       = "5601"
-
 default['kibana']['systemd']                    = "true"
-default['kibana']['home']                       = node['hopslog']['dir'] + "/kibana-" + "#{node['kibana']['version']}-linux-x86_64"
+default['kibana']['home']                       = node['hopslog']['dir'] + "/opensearch-dashboards-#{node['kibana']['version']}"
 default['kibana']['base_dir']                   = node['hopslog']['dir'] + "/kibana"
```
Comment (maismail) [inline attributes/default.rb:48]: shouldn't we change the base_dir to `opensearch-dashboards`?

**Review by maismail — CHANGES_REQUESTED** [review]

File: attributes/default.rb (hunk 1 above)
Comment (jimdowling) [reply attributes/default.rb:17]: I want to leave it there as a reminder that some day we should validate the file hashes...

File: attributes/default.rb (hunk 10 above)
Comment (jimdowling) [reply attributes/default.rb:48]: I didn't want to break any dependencies on it, but i guess i should just find any.

File: templates/default/opensearch_dashboards.yml.erb (hunk 6 above)
Comment (jimdowling) [reply templates/default/opensearch_dashboards.yml.erb]: Done.

File: templates/default/start-kibana.sh.erb (hunk 9 above)
Comment (jimdowling) [reply templates/default/start-kibana.sh.erb]: I changed it to  opensearch-dashboards - not underscore and plural for dashboards. You don't use underscore in service names.

**Review by o-alex — APPROVED** [review]

File: templates/default/start-kibana.sh.erb (hunk 8 above)
Comment (jimdowling) [reply templates/default/start-kibana.sh.erb]: ack

**Review by maismail — APPROVED** [review]

**Pull request merged**